### PR TITLE
Formatting and commenting

### PR DESCRIPTION
**Pull request by peschkaj**
Provides better comments and re-formats code to be more idiomatic C#.

WIP on #38

### CODE REVIEW

**Comment by peschkaj** [comment]
These conflicts are almost certainly related to Connor's big reorg and will be fixed when you check out.

**Comment by connorkuehl** [comment]
`git blame connor` :-(

**Comment by peschkaj** [comment]
LOL, not actually your fault, but just because git gets confused with renames sometimes. I should've merged your changes first. Shows what I know!

File: Assets/Scripts/LevelManager.cs [hunk 1]
```diff
@@ -1,138 +1,138 @@
...
+                _gridPositions.Add(new Vector3(x, y, 0f));
             }
         }
     }
 
-
-    //Sets up the outer walls and floor (background) of the game board.
-    void BoardSetup()
+    /// <summary>
+    /// Sets up the outer walls and floor (backgroud) of the game board.
```
Comment (connorkuehl) [inline Assets/Scripts/LevelManager.cs]: Background*

File: Assets/Scripts/LevelManager.cs [hunk 2]
```diff
@@ -1,138 +1,138 @@
...
         }
     }
 
-
-    //RandomPosition returns a random position from our list gridPositions.
-    Vector3 RandomPosition()
+    /// <summary>
+    /// RandomPosition returns a random position from our list gridPositions.
+    /// </summary>
+    /// <returns></returns>
```
Comment (connorkuehl) [inline Assets/Scripts/LevelManager.cs]: Is this a vestigial tag? The method name is self-documenting, but if we're going to use the tag we might as well go all the way with it and put a blurb in between the tags.

File: Assets/Scripts/LevelManager.cs [hunk 3]
```diff
@@ -143,16 +143,23 @@ public void SetupScene(int level)
...
-
-
-
 
+    public class Range
+    {
+        public int Max;
+        public int Min;
 
+        private Range(int min, int max)
```
Comment (connorkuehl) [inline Assets/Scripts/LevelManager.cs]: `<summary>` or nah? I'd say this class is simple and self-documenting enough.

File: Assets/Scripts/MovingObject.cs [hunk 4]
```diff
@@ -1,119 +1,139 @@
 ﻿using System.Collections;
-using System.Collections.Generic;
 using UnityEngine;
 
-//The abstract keyword enables you to create classes and class members that are incomplete and must be implemented in a derived class.
+/// <summary>
+///     An object, that moves.
+/// </summary>
+/// <remarks>Since this is an abstract class, it requires some actual implementation</remarks>
```
Comment (connorkuehl) [inline Assets/Scripts/MovingObject.cs]: Duh

File: Assets/Scripts/MovingObject.cs [hunk 5]
```diff
@@ -1,119 +1,139 @@
...
 
-        //If something was hit, return false, Move was unsuccesful.
+        // If something was hit, return false, Move was unsuccesful.
         return false;
     }
 
 
-    //Co-routine for moving units from one space to next, takes a parameter end to specify where to move to.
+    /// <summary>
+    ///     Co-routine for moving units from one space to next, takes a parameter end to specify where to move to.
```
Comment (connorkuehl) [inline Assets/Scripts/MovingObject.cs]: I think the last thought in this sentence would be better placed between the `param` tags.

File: Assets/Scripts/LevelManager.cs (hunk 1 above)
Comment (peschkaj) [inline Assets/Scripts/LevelManager.cs]: 👍

File: Assets/Scripts/LevelManager.cs (hunk 2 above)
Comment (peschkaj) [inline Assets/Scripts/LevelManager.cs]: The tag was auto-added. I've populated it.

File: Assets/Scripts/LevelManager.cs (hunk 3 above)
Comment (peschkaj) [inline Assets/Scripts/LevelManager.cs]: Added a summary anyway.

File: Assets/Scripts/MovingObject.cs (hunk 5 above)
Comment (peschkaj) [inline Assets/Scripts/MovingObject.cs]: :+1:

**Pull request merged**